src/common/utxobased/engine/makeUtxoEngineState.ts

```diff
@@ -397,7 +397,7 @@ interface RawUtxoCache {
   [key: string]: {
     processing: boolean
     path: ShortPath
-    address: Required<IAddress>
+    address: IAddress
```

Comment:
I'm not sure what benefit the `Required` field had here either. It doesn't really improve all that much, since we have to check the optional fields anyway. There is a potential side-task after this though, because once we add a redeem script field to an IAddress, there is no reason for keeping a `path` field anymore. The sole reason for the `path` field is that we can derive a redeem script on the fly. If we derive the redeem script when we derive the address, there is no need for that anymore.

Reply:
I did this addiional derivation step in 9acc198fcfed6b1959e9373f2bd28e7ff9bf067c